pkg/gadgettracermanager/gadgettracermanager_test.go

```diff
@@ -0,0 +1,218 @@
+				},
+			},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+				Labels: []*pb.Label{
+					&pb.Label{Key: "key1", Value: "value1"},
+					&pb.Label{Key: "key2", Value: "value2"},
+				},
+			},
+		},
+		{
+			description: "Podname does not match",
+			match:       false,
+			selector: &pb.ContainerSelector{
+				Namespace: "this-namespace",
+				Podname:   "this-pod",
+			},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "a-misnamed-pod",
+				ContainerName: "this-container",
+			},
+		},
+		{
+			description: "One label doesn't match",
+			match:       false,
+			selector: &pb.ContainerSelector{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+				Labels: []*pb.Label{
+					&pb.Label{Key: "key1", Value: "value1"},
+					&pb.Label{Key: "key2", Value: "value2"},
+				},
+			},
+			container: &pb.ContainerDefinition{
+				Namespace:     "this-namespace",
+				Podname:       "this-pod",
+				ContainerName: "this-container",
+				Labels: []*pb.Label{
+					&pb.Label{Key: "key1", Value: "value1"},
+					&pb.Label{Key: "key2", Value: "something-else"},
+				},
+			},
+		},
+	}
+
+	for i, entry := range table {
+		result := containerSelectorMatches(entry.selector, entry.container)
+		if entry.match != result {
+			t.Fatalf("Failed test %q (index %d): result %v expected %v",
+				entry.description, i, result, entry.match)
+		}
+	}
+}
+
+func TestTracer(t *testing.T) {
+	g, err := newServer("fake-node", false, false, false)
+	if err != nil {
+		t.Fatalf("Failed to create new server: %v", err)
+	}
+
+	ctx := context.TODO()
+
+	// Add 3 Tracers
+	for i := 0; i < 3; i++ {
+		respTrace, err := g.AddTracer(ctx, &pb.AddTracerRequest{
+			Id: fmt.Sprintf("my_trace_id%d", i),
+			Selector: &pb.ContainerSelector{
+				Namespace: fmt.Sprintf("this-namespace%d", i),
+			},
+		})
+		if err != nil {
+			t.Fatalf("Failed to add trace: %v", err)
+		}
+		if respTrace.Id != fmt.Sprintf("my_trace_id%d", i) {
+			t.Fatalf("Error while adding trace: got id %q", respTrace.Id)
+		}
+	}
+
+	if len(g.tracers) != 3 {
+		t.Fatalf("Error while checking tracers: len %d", len(g.tracers))
+	}
+
+	// Remove 1 Tracer
+	respTrace, err := g.RemoveTracer(ctx, &pb.TracerID{
+		Id: fmt.Sprintf("my_trace_id%d", 1),
+	})
+	if err != nil {
+		t.Fatalf("Failed to remove tracer: %v", err)
+	}
+	if respTrace == nil {
+		t.Fatal("Error while removing tracer: invalid response")
+	}
+
+	// Check content
+	if len(g.tracers) != 2 {
+		t.Fatalf("Error while checking tracers: len %d", len(g.tracers))
+	}
+	_, ok := g.tracers["my_trace_id0"]
+	if !ok {
+		t.Fatalf("Error while checking tracer %s: not found", "my_trace_id0")
+	}
+	_, ok = g.tracers["my_trace_id2"]
+	if !ok {
+		t.Fatalf("Error while checking tracer %s: not found", "my_trace_id2")
+	}
```

Comment:
Some ideas for more test cases:
- try to add an already existing tracer
- remove a non existing tracer